### PR TITLE
Expose CTranslate2::ctranslate2 as CMake package

### PR DESCRIPTION
**Pull request by HennerM**
Thanks for creating this package!

CTranslate2 already provides an easy to use pip package and has a nice C++ interface. 
However, it wasn't that easy to consume that in another C++ project, this is something I plan to use for a ctranslate2 backend for https://developer.nvidia.com/nvidia-triton-inference-server. (which I am planning to open-source soon).

Most of the changes in here just add the CMake config so that it can be used with the `find_package` command. 
I based this on the CMake guide here, https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html. I am not that experienced with CMake, but it seems to work in my other project 🤷

### CODE REVIEW

**Review by guillaumekln — COMMENTED** [review]
Thanks! This seems useful. I have some comments:

File: CMakeLists.txt [hunk 1]
```diff
@@ -51,6 +51,12 @@ endif(NOT CMAKE_BUILD_TYPE)
 # Set CXX flags.
 set(CMAKE_CXX_STANDARD 17)
 
+execute_process(COMMAND python3 -c "from ctranslate2.version import __version__; print(__version__, end=\'\')"
```
[the remark this answers is not in the document]
Comment (HennerM) [reply CMakeLists.txt]: Good point. I now parse the version.py in CMake. Admittedly it's not pretty, but it works 🤷

File: CMakeLists.txt [hunk 2]
```diff
@@ -51,6 +51,12 @@ endif(NOT CMAKE_BUILD_TYPE)
...
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/python
+  OUTPUT_VARIABLE CTRANSLATE2_VERSION
+)
+
+message(${CTRANSLATE2_VERSION})
```
[the remark this answers is not in the document]
Comment (HennerM) [reply CMakeLists.txt]: I actually just left this in for debugging, removed now

**Review by guillaumekln — APPROVED** [review]
This looks good to me. Thanks!

Have you pushed all changes?

**Comment by HennerM** [comment]
> This looks good to me. Thanks!
> 
> Have you pushed all changes?

Thanks, yes it's good to go in. FYI here is the project I am working on that is using CTranslate2 as a library: https://github.com/speechmatics/ctranslate2_triton_backend

**Pull request merged**